frontend/src/types/Graph.ts

```diff
@@ -513,6 +512,10 @@ export interface DecoratedGraphEdgeData extends GraphEdgeData {
   tcp: number;
   // Default value NaN
   throughput: number;
+  // Waypoint from/to
+  waypoint?: string;
+  // Waypoint edge
+  waypointEdge?: DecoratedGraphEdgeData;
```

Comment:
This seems like a circular definition, but I guess it works.  What is happening with this "waypointEdge" is pretty tricky, maybe we could use some comments somewhere about how we're doing bidirectional edges.